index.js

```diff
@@ -339,3 +339,69 @@ module.exports.random = {
     return s * Math.pow(2, (-64 - e))
   }
 }
+
+/**
+ * A dictionary of header values, i.e '{ 'header-name': 'header-value' }'
+ * @typedef {Object.<string, (string | string[] | undefined)>} HeaderLike
+ */
+
+/**
+ * Uses Ed25519: a public-key signature system {@link https://ed25519.cr.yp.to/} {@link https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12}
+ * Signs the message using the secret key and returns a signature.
+ * @see {nacl.sign.detached}
+ *
+ * @method
+ * @param {string} keyId - an opaque string that the server can use to look up the component they need to validate the signature
+ * @param {string} secretKey - secret key to sign the message
+ * @param {HeaderLike} headers - headers containing the properties to sign
+ * @returns {string}
+ */
+module.exports.ed25519Sign = function (keyId /* string */, secretKey /* string */, headers = {} /* dictionary */) {
+  if (!secretKey) throw new Error('secret key is required')
+  if (!keyId) throw new Error('key id is required')
+  if (Object.keys(headers).length === 0) throw new Error('headers are required')
+
+  const headerKeys = []
+  const message = Object.entries(headers)
```

Comment:
It looks like this will return the keys in sorted order (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries). However I don't think the brave-core version does anything like that: https://github.com/brave/brave-core/blob/a523199e4702e4907908a09969d023f5725402dc/components/brave_rewards/core/common/request_signer.cc#L99

I wonder if this means we'll get different signatures for headers like `{b: 1, a: 2}` in brave-core versus this, because they will be serialized differently.

Reply:
I don't believe Object.entries should sort anything, unless I am mis-understanding. If the browser gives us
`{b: 1, a: 2}`

we should give `headers="b a"` back.

Reply:
it seems Object.entries does have a concept of order https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582 

so that explains the MDN example:

```
const randomKeyOrder = { 100: "a", 2: "b", 7: "c" };
console.log(Object.entries(randomKeyOrder)); // [ ['2', 'b'], ['7', 'c'], ['100', 'a'] ]
```

i'm guessing we won't run into this because the header keys should always be strings, which are in insertion order (same as in C++ hopefully)